src/bin/pgcopydb/snapshot.c

```diff
@@ -230,7 +229,7 @@ copydb_close_snapshot(CopyDataSpec *copySpecs)
 			/* only COMMIT sql snapshot kinds, no need for logical rep ones */
 			if (!pgsql_commit(pgsql))
 			{
-				char pguri[MAXCONNINFO] = { 0 };
+				char *pguri = (char *) calloc(strlen(snapshot->pguri) + 1, sizeof(char));
```

Comment:
Why is that necessary? The `parse_and_scrub_connection_string` calls into `buildPostgresURIfromPieces` which should be responsible for memory allocation like discussed earlier. Just initialize to NULL here. And make sure to `free` the allocated memory.

Reply:
The function docs for `buildPostgresURIfromPieces` says it's the caller's responsibility to allocate pguri to atleast 1024 bytes which is what I did here as well and so at other places where `parse_and_scrub_connection_string` is being called. This is in follow to the conversation above where sageURI.pguri or scrubbedURI are working with just MAXXCONNINFO length because the passwords are masked.

Reply:
Well the comment there applies to code before your PR, when we used to use static memory allocation (string buffers) for connection strings. It's not the case anymore, so that needs to change. Also we can't approach this PR with having in mind that only the password field is going to be a problem. For instance, we build a new Postgres URI to connect in logical replication mode, and we need our internal facilities to be able to handle that.